### PR TITLE
Handle multiple diagnostic errors in a single `expectError` assertion

### LINKED ISSUE #87: expectError on array arguments mismatch
Following example:
```typescript

interface A {
  test1(a: '*'): string;
  test1(a: 'a' | 'b'): string;
  test1(a: ReadonlyArray<'a' | 'b'>): string;
  test1(a: never): string;
}
const a: A = null as any;

expectError(a.test1(['a', 'bad']));

```

```

No overload matches this call.
  The last overload gave the following error.
    Type string is not assignable to type never.
      Type string is not assignable to type never. 
```

### PR DESCRIPTION
**Pull request by BendingBender**
Fixes #87

### CODE REVIEW

**Review by sindresorhus — APPROVED** [review]

File: source/lib/compiler.ts [hunk 1]
```diff
@@ -28,21 +28,26 @@ const expectErrordiagnosticCodesToIgnore = new Set<DiagnosticCode>([
...
  *
  * @param diagnostic - The diagnostic to validate.
  * @param expectedErrors - Map of the expected errors.
  * @returns Boolean indicating if the diagnostic should be ignored or not.
  */
-const ignoreDiagnostic = (diagnostic: TSDiagnostic, expectedErrors: Map<Location, any>): boolean => {
+const ignoreDiagnostic = (
+	diagnostic: TSDiagnostic,
+	expectedErrors: Map<Location, ExpectedError>
+): IgnoreDiagnosticResult => {
```
Comment (SamVerschueren) [inline source/lib/compiler.ts:43]: Can you update the `@returns` in the TS doc?

File: source/lib/compiler.ts [hunk 2]
```diff
@@ -80,9 +83,19 @@ export const getDiagnostics = (context: Context): Diagnostic[] => {
...
+	const expectedErrorsLocationsWithFoundDiagnostics: Location[] = [];
 
 	for (const diagnostic of tsDiagnostics) {
-		if (!diagnostic.file || ignoreDiagnostic(diagnostic, expectedErrors)) {
+		if (!diagnostic.file) {
+			continue;
+		}
+
+		const ignoreDiagnosticResult = ignoreDiagnostic(diagnostic, expectedErrors);
+		if (ignoreDiagnosticResult !== 'preserve') {
```
Comment (SamVerschueren) [inline source/lib/compiler.ts:94]:
```suggestion
		const ignoreDiagnosticResult = ignoreDiagnostic(diagnostic, expectedErrors);

		if (ignoreDiagnosticResult !== 'preserve') {
```

**Review by SamVerschueren — CHANGES_REQUESTED** [review]
Two very small comments. Looks good! Thanks for working on this 🙏 !

File: source/lib/compiler.ts (hunk 2 above)
Comment (BendingBender) [reply source/lib/compiler.ts]: done

File: source/lib/compiler.ts (hunk 1 above)
Comment (BendingBender) [reply source/lib/compiler.ts:45]: done

**Review by SamVerschueren — APPROVED** [review]

**Comment by SamVerschueren** [comment]
Awesome work @BendingBender 🙏 !

**Pull request merged**